src/Integration/Binding/CSharpVBProjectBinder.cs

```diff
@@ -108,15 +126,20 @@ private bool IsSolutionBound(BindingConfiguration binding, Language language, ou
             solutionRuleSetFilePath = null;
             additionalFilePath = CSharpVBBindingConfigProvider.GetSolutionAdditionalFilePath(language, binding);
 
-            if (!fileSystem.File.Exists(additionalFilePath))
+            var additionalFileExists = fileSystem.File.Exists(additionalFilePath);
+            logger.LogDebug($"[Binding check] Solution-level additional file exists: {additionalFileExists} (file path: '{additionalFilePath}')");
+            if (!additionalFileExists)
             {
                 return false;
             }
 
             var solutionRuleSet = GetSolutionRuleSet(binding, language);
             solutionRuleSetFilePath = solutionRuleSet?.FilePath;
 
-            return solutionRuleSet != null;
+            var rulesetExists = solutionRuleSet != null;
+            logger.LogDebug($"[Binding check] Solution-level ruleset exists: {rulesetExists} (file path: '{solutionRuleSetFilePath}')");
+
+            return rulesetExists;
         }
 
         private RuleSet GetSolutionRuleSet(BindingConfiguration binding, Language language)
```

Comment:
It might be interesting to add logging in `ruleSetSerializer.LoadRuleSet` -- at the moment I see that it catches exceptions and writes them into `Trace.WriteLine`.

Reply:
Ok. I'll drop the Trace statement and pass a logger to the serializer.